lib/functional-constraints/requiredSamples.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const _ = require('lodash');
+const jsonschema = require('jsonschema');
+
+// todo: deal with circular dep.
+const RESOURCE_ID = '/ResourceSchema';
+const RESOURCE_METHODS = ['get', 'hook', 'list', 'search', 'create'];
+
+const check = definition => {
+  if (!definition.operation || _.get(definition, 'display.hidden')) return null;
+
+  const samples = _.get(definition, 'operation.sample', {});
+  if (!Object.keys(samples).length) {
+    return new jsonschema.ValidationError(
+      'required "sample" for non-hidden operation',
```

Comment:
I think this gets phrased weird to the end user. Maybe `... requires "sample", because it's not hidden`